src/aspire/basis/ffb_2d.py

```diff
@@ -281,7 +296,7 @@ def expand(self, x):
         for isample in range(0, n_data):
```

Comment:
The `expand` method has been moved to the base class in a recent commit. Probably we should merge with the develop branch first.

Reply:
Sure, go ahead. The tests should pass with both libraries. Thanks

Reply:
I have merged the code from the develop  branch. Please solve the conflicts and push back so that  we can merge the PR.

Reply:
Junchao, I think there is a miss understanding.  I was suggesting maybe you go ahead take point on melding your commits from this morning, not just merge the code in a shared branch willy nilly. I suggest this because in the recent PR I pointed out this method was a problem, and not using our (nfft) framework Vineet created. The one we use elsewhere. However, you rebutted the concern, and it was quickly rushed into dev, master, and now the hackathon branch.  Now we do have a conflict, yes.

I don't really think that was the most productive way to handle that or meld our work.  If you will have trouble to resolve I can take this, but it will take me away from my work, again.  I have already spent a lot of time helping you to prepare for the hackathon and it is beginning to be very disruptive to me getting things done.  In fact, this very work here was totally halted so I could help you.

I'm not sure there is any direct benefit to having the hackathon branch be on the HEAD at this point; in which case we can avoid the conflict until later, when the code is prepared for production. Do you agree? If so you might revert.

Thanks.

Reply:
The expend ffb_2d.py was converted by me from Matlab code when I just joined the group. It did use the Vineet's framework. But it was reorganized into the base class for a better structure recently. This update is related to our Hackathon. So it better to include.  Other code are not related such as the new Polar2D basis which use the many frame and we don't need to do anything. This is only a minor issue.

Reply:
Right, but I am not going to half fix it just to cover the few conflict lines.  If I go back and deal with those changes, that effort should fix them everywhere in dev (and master now it seems), which would include the work that was just rolled in. That is the right thing to do.  There is too much adhoc...

I feel very slighted that you have taken advantage of the extra time I have returned to you (in a sense) by cleaning/setting up hackathon efforts etc, from my work and personal time, and used it to rush code in, which you know would have a conflict, the Friday before a kick off meeting. That is not a team friendly move.  My work very well could have been merged in first if I had ignored collaborating with you or others, or ignored the consequences to your efforts and focused only on my own results. There was nothing stopping me from clicking the buttons, or not giving the time to what I considered team goals.  That's not the type of developer I want or need to be at this stage in my career.  I am hoping that we can use this and some of the other hackathon tensions as an example of ways small actions can be either pro/counter productive before we get into a real sprint together.  I think communicating what we are working on in a regular (asynch is fine) fashion is going to be critical.

I am going to take this task, monday, because I know the Hackathon has put some pressure on you, and that you are just trying to finish up some work first. It is small, but it is never the amount of code :) That's not the conflict I worry about.  This needs to improve though.

Take it easy, we're in a good spot! Have a nice weekend, we will talk soon.

Reply:
I have solved the conflict in five minutes. I understand both of us have to be busy on preparing the Hackathon event. We can have a better communication through phone if you like.